pygradle-plugin/src/main/groovy/com/linkedin/gradle/python/extension/ZipappExtension.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.gradle.python.extension;
+
+import com.linkedin.gradle.python.util.OperatingSystem;
+
+import java.io.File;
+
+
+public class ZipappExtension {
```

Comment:
I think this would should be an interface vs a class. We can make an abstract class to hold some helpers if needed.

Reply:
I made `ZipappExtension` a class to hold the common fields `cache` and `isFat` which are used in both the pex extension subclass, and the (currently only available in a separate non-public extension) shiv extension subclass.  If we make `ZipappExtension` an interface, where will we put `cache` and `isFat`?  Also, I think this class is more about providing common functionality for subclasses rather than defining a useful interface that other classes should implement.  E.g., it's all about getting and setting the cache directory and the "fatness" of the artifacts.

Reply:
@kopy07 Fair enough.  I think now that we have the `cache` property sorted out, I agree with you about making `ZipappExtension` an interface.  The next push will do that.